pkg/controller/logstorage/eskubecontrollers.go

```diff
@@ -88,26 +85,22 @@ func (r *ReconcileLogStorage) createEsKubeControllers(
 
 	imageSet, err := imageset.GetImageSet(ctx, r.client, install.Variant)
 	if err != nil {
-		reqLogger.Error(err, "Error getting ImageSet")
-		r.status.SetDegraded("Error getting ImageSet", err.Error())
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Error getting ImageSet", err, reqLogger)
 		return reconcile.Result{}, false, err
 	}
 
 	if err = imageset.ValidateImageSet(imageSet); err != nil {
-		reqLogger.Error(err, "Error validating ImageSet")
-		r.status.SetDegraded("Error validating ImageSet", err.Error())
+		r.status.SetDegraded(operatorv1.ResourceValidationError, "Error validating ImageSet", err, reqLogger)
 		return reconcile.Result{}, false, err
 	}
 
 	if err = imageset.ResolveImages(imageSet, esKubeControllerComponents); err != nil {
-		reqLogger.Error(err, "Error resolving ImageSet for elasticsearch kube-controllers components")
-		r.status.SetDegraded("Error resolving ImageSet for elasticsearch kube-controllers components", err.Error())
+		r.status.SetDegraded(operatorv1.ResourceUpdateError, "Error resolving ImageSet for elasticsearch kube-controllers components", err, reqLogger)
```

Comment:
I think this is also a ResourceValidationError

Reply:
Done